src/components/AndelaCentres/AndelaCentresComponent.jsx

```diff
@@ -32,18 +38,78 @@ class AndelaCentresComponent extends React.Component {
     this.props.loadOfficeLocations(activePage);
   };
 
+  handleToggleModal = (id) => {
+    this.setState({
+      modalOpen: !this.state.modalOpen,
+      locationId: id
+    });
+    this.props.resetMessage();
+  }
+
+  onSelectCountry = (event, data) => {
+    this.setState({ country: data.value });
+  }
+
+  handleChange = (event) => {
+    const { value, name } = event.target;
+
+    this.setState({
+      [name]: value
+    });
+  }
+
+  handleEditSubmit = (event) => {
+    event.preventDefault();
+
+    this.props.updateAndelaCentre(this.state.locationId, {
+      centre_name: this.state.centre,
+      country: this.state.country
+    }).then(() => {
+      this.props.loadOfficeLocations(this.state.activePage);
+    });
+  };
+
   getTotalPages = () => Math.ceil(this.props.locationCount / this.state.limit);
 
   render() {
-    const { isLoading, locationList, error, resetMessage } = this.props;
+    const {
+      isLoading,
+      locationList,
+      error,
+      resetMessage,
+      entity,
+      updateSuccess,
+      updateError
+    } = this.props;
     const hasLocations = !isEmpty(locationList);
     const showStatus = error;
+    const showAction = entity === 'andela-centres';
     const showNotFound = !isLoading && !hasLocations && !showStatus;
 
     return (
       <NavBarComponent>
         <PageHeader header="Andela Centres" />
 
+        {this.state.modalOpen &&
+          <ModalComponent
+            modalTitle="Update An Andela Centre"
+            toggleModal={this.handleToggleModal}
+            modalOpen={this.state.modalOpen}
+          >
+            <CentreModal
```

Comment:
You are passing more than 8 props here. Remember we are [limiting the number to 8](https://github.com/AndelaOSP/art-dashboard/blob/develop/docs/best_practices/conventions.md#limit-props-being-passed-to-child-component)

You can fix that by:
1. creating a [container](https://github.com/AndelaOSP/art-dashboard/blob/develop/docs/best_practices/code_organization.md#_components-folder) for `CenterModal` in `_components/AndelaCenters/..`
2. Turning `CenterModal` into a class and transfer the methods such as `handleChange` to it

OR you can restructure the code the way you deem fit. The decision is yours